src/components/game/GameEngine.test.js

```diff
@@ -351,6 +352,27 @@ describe('GameEngine Component with IndexedDB', () => {
     jest.useRealTimers();
   });
 
+  it('Check that restart button can be used to restart game', async () => {
+    jest.useFakeTimers();
+
+    render(
+      <MemoryRouter>
+        <GameEngine pathId={String(pathId)} />
+        <GameEndingPage />
+      </MemoryRouter>
+    );
+
+    fireEvent.click(screen.getByRole('button', { name: /Restart Game/i }));
+
+    render(
+      <MemoryRouter>
+        <GameEngine pathId={String(pathId)} />
+      </MemoryRouter>
+    );
```

Comment:
Tätä ei kai pitäisi uudelleen renderöidä? Varmaan samoja vikoja kuin toisessa testissä, eli se ei kykene testissä navigoimaan taaksepäin koska eteenpäin ei ole ikinä navigoitu.